### PR TITLE
mfa though

### PR DESCRIPTION
**Pull request by rebstar6**
3rd times the charm?

### CODE REVIEW

File: modules/portal/app/controllers/OidcAuthenticator.scala [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+        "response_type" -> responseType,
+        "redirect_uri" -> redirectUri,
+        "scope" -> "openid profile email")
+
+    (call.url, call.queryString)
+  }
+
+  def isNotExpired(claimsSet: JWTClaimsSet): Boolean = {
+    val now = new Date()
+    val expirationTime = claimsSet.getExpirationTime
```
Comment (nimaeskandary) [inline modules/portal/app/controllers/OidcAuthenticator.scala]: is this an optional? according to https://tools.ietf.org/html/rfc7519#section-4.1 seems the claimset doesn't need to include it

File: modules/portal/app/actions/FrontendAction.scala [hunk 2]
```diff
@@ -50,6 +55,6 @@ class FrontendAction(val userLookup: String => IO[Option[User]])(
 
   def lockedUserResult(un: String): Future[Result] =
     Future.successful(
-      Redirect("/noaccess")
+      Redirect("/noaccess").withNewSession
```
Comment (britneywright) [inline modules/portal/app/actions/FrontendAction.scala]: if we add `withNewSession` here they'll be sent back through the authentication process if they try to go to a another page in the portal, vs continuing to be sent to the noaccess page.

File: modules/portal/app/actions/VinylDnsAction.scala [hunk 3]
```diff
@@ -17,31 +17,68 @@
...
+        logger.info("User is not logged in or token expired; redirecting to login screen")
+        notLoggedInResult
 
       case Some(un) =>
         // user name in session, let's get it from the repo
         userLookup(un).unsafeToFuture().flatMap {
           // Odd case, but let's handle with a different error message
-          case None => cantFindAccountResult(un)
+          case None =>
+            logger.error(s"Cant find account for user  with username $un")
```
Comment (britneywright) [inline modules/portal/app/actions/VinylDnsAction.scala]: there's an extra space here between "user" and "with"

File: modules/portal/app/actions/FrontendAction.scala (hunk 2 above)
Comment (nimaeskandary) [reply modules/portal/app/actions/FrontendAction.scala]: hm, now that I think about it the no access page breaks the logout flow no? Could we potentially make an issue to add a link to `/logout` on the no access screen and then this `.withNewSession` could be removed?

Comment (nimaeskandary) [reply modules/portal/app/actions/FrontendAction.scala]: Or maybe add that link before this pr goes in @rebstar6 😛

Comment (britneywright) [reply modules/portal/app/actions/FrontendAction.scala]: I don't think we need that. it's fine for the user to remain logged in and locked out. they'll always wind up on the no access screen so we're saving them the effort of logging in over and over to only be redirected to the no access page. If their account were unlocked they'd have access again.

Comment (nimaeskandary) [reply modules/portal/app/actions/FrontendAction.scala]: Hm, I don't see a reason to not let the logged in user log out at all if they want, keeping them locked into their account in their browser seems like an issue

Comment (rebstar6) [reply modules/portal/app/actions/FrontendAction.scala]: i think its ok as is, logout was an extra add anyway

**Pull request merged**